src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -123,6 +188,57 @@ export const Configurator = ({
           >
             <EuiFieldText disabled value={dataset.title} />
           </EuiFormRow>
+          {indexedViewsService && (
+            <>
+              <EuiSpacer />
+              <EuiSwitch
+                compressed
+                checked={shouldSelectIndexedView}
+                label={
+                  <EuiFormLabel>
+                    {i18n.translate(
+                      'data.explorer.datasetSelector.advancedSelector.configurator.showAvailableIndexedViewsLabel',
+                      {
+                        defaultMessage: 'Query indexed view',
+                      }
+                    )}
+                  </EuiFormLabel>
+                }
+                onChange={(e) => setShouldSelectIndexedView(e.target.checked)}
+              />
+              <EuiSpacer size="m" />
+              {shouldSelectIndexedView && (
+                <EuiFormRow
+                  label={i18n.translate(
+                    'data.explorer.datasetSelector.advancedSelector.configurator.indexedViewLabel',
+                    {
+                      defaultMessage: 'Available indexed views',
+                    }
+                  )}
+                  helpText={i18n.translate(
+                    'data.explorer.datasetSelector.advancedSelector.configurator.indexedViewHelpText',
+                    {
+                      defaultMessage: 'Select an indexed view to speed up your query.',
+                    }
+                  )}
+                >
+                  <EuiSelect
+                    isLoading={isLoadingIndexedViews}
+                    options={indexedViews.map(({ name }) => ({
+                      text: name,
```

Comment:
is there no option to select an option and then unselect it if the user did not want to have an indexed view? similar to how the timefieldname provides an option. we should follow the same UX for consistency

Reply:
Adding in a followup